frontend/src/components/assessments/student-experience/NavButtons.tsx

```diff
@@ -1,15 +1,53 @@
 import React, { useContext } from "react";
+import { useMutation } from "@apollo/client";
 import { Button, HStack, Spacer } from "@chakra-ui/react";
 
+import { SUBMIT_TEST } from "../../../APIClients/mutations/TestSessionMutations";
+import AuthContext from "../../../contexts/AuthContext";
 import StudentContext from "../../../contexts/StudentContext";
+import { mapAnswersToResultsArray } from "../../../utils/StudentUtils";
+import Toast from "../../common/Toast";
 
 const NavButtons = (): React.ReactElement => {
-  const { test, currentQuestionIndex, setCurrentQuestionIndex } = useContext(
-    StudentContext,
-  );
+  const {
+    answers,
+    test,
+    testSession,
+    currentQuestionIndex,
+    setCurrentQuestionIndex,
+  } = useContext(StudentContext);
+  const { showToast } = Toast();
+  const { authenticatedUser } = useContext(AuthContext);
 
-  const questionCount = test?.questions.length ?? 0;
+  const [submitTest, { error }] = useMutation<{
+    submitTest: string;
+  }>(SUBMIT_TEST);
+
+  const handleSubmitTest = async () => {
+    await submitTest({
+      variables: {
+        id: testSession?.id,
+        result: {
+          student: authenticatedUser?.id,
+          answers: mapAnswersToResultsArray(answers),
+        },
+      },
+    });
+    if (error) {
```

Comment:
Will this show an error as expected? Usually when you await something it will throw upon error, not sure about Apollo though...